### PR TITLE
Ionic documentation added

### PR DESCRIPTION
**Pull request by nevercode-kevin**
Proposed documentation for building **Ionic Capacitor** and **Ionic Cordova** apps.

### CODE REVIEW

**Comment by VeArnold** [comment]
Might be better to change instance type to mac mini? If users do not pay enough attention, then might cause issues :)

File: content/getting-started/building-an-ionic-app.md [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+
+The apps you have available on Codemagic are listed on the Applications page. See how to add additional apps [here](./adding-apps-from-custom-sources).
+
+1. On the Applications page, click **Set up build** next to the app you want to start building. 
+2. Select **Ionic App** as the starter workflow.
+3. Download the example configuration for Ionic App or copy it to clipboard.
+4. Then edit the configuration file to adjust it to your project needs and commit it to the root of your repository.
+    * For an overview about using `codemagic.yaml`, please refer [here](./yaml). 
+    * Read more about adding configuration for [testing](../testing-yaml/testing), [code signing](../code-signing-yaml/signing) and [publishing](../publishing-yaml/distribution).
+    * See the full Android workflow example [below](#android-workflow-example).
```
Comment (helinanever) [inline content/getting-started/building-an-ionic-app.md:19]: Should be something along the lines of "See the full workflow examples below" and no link.

File: content/getting-started/building-an-ionic-app.md [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+
+## Testing, code signing and publishing Ionic Android and iOS apps
+
+To test, code sign and publish Ionic Android and iOS apps:
+
+* The code for testing an Ionic Android app also goes under `scripts`. A few examples of testing can be found [here](../testing-yaml/testing).
+* All Android applications need to be signed before release. For Gradle code signing configuration for **Ionic Capacitor** apps refer to the [documentation](../code-signing/android-code-signing/). More information about code signing with YAML in general is [here](../code-signing-yaml/signing). If you are building **Ionic Cordova** Android apps see the workflow sample below.
+* All iOS applications need to be signed before release. For iOS code signing configuration refer to the iOS code signing documentation [here](../code-signing/ios-code-signing/)
+* All generated artifacts can be published to external services. The available integrations currently are email, Slack and Google Play. It is also possible to publish elsewhere with custom scripts (e.g. Firebase App Distribution). Script examples for all of them are available [here](../publishing-yaml/distribution/#publishing).
+
```
Comment (helinanever) [inline content/getting-started/building-an-ionic-app.md:39]: the link should be: ../publishing-yaml/distribution/

File: content/getting-started/building-an-ionic-app.md [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+            #     echo ${CM_PROVISIONING_PROFILE} | base64 --decode > $PROFILE_PATH
+            #     echo "Saved provisioning profile $PROFILE_PATH"
+            - name: Fetch signing files
+              script: |
+                # app-store-connect fetch-signing-files "com.nevercode.ncionicapp" --type IOS_APP_STORE --create
+                app-store-connect fetch-signing-files $(xcode-project detect-bundle-id) --type IOS_APP_STORE --create
+            # - name: Set up signing certificate (Use with manual code signing)
+            #   script: |
+            #     echo $CM_CERTIFICATE | base64 --decode > /tmp/certificate.p12
+            #     keychain add-certificates --certificate /tmp/certificate.p12 --certificate-password $CM_CERTIFICATE_PASSWORD
```
Comment (helinanever) [inline content/getting-started/building-an-ionic-app.md:251]: I know you want to explicitly show two code signing options but having the two options in the file and some lines marked as commentary creates a lot of noise and makes the file difficult to read

File: content/getting-started/building-an-ionic-app.md [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
```
Comment (helinanever) [inline content/getting-started/building-an-ionic-app.md:357]: Same comment here about code signing

**Review by helinanever — CHANGES_REQUESTED** [review]

**Review by helinanever — APPROVED** [review]
LGTM

**Pull request merged**